mlir/include/mlir/Dialect/LLVMIR/LLVMOps.td

```diff
@@ -508,6 +508,23 @@ class LLVM_CastOp<string mnemonic, string instName, Type type,
       $_location, $_resultType, $arg);
   }];
 }
+class LLVM_CastOpWithOverflowFlag<string mnemonic, string instName, Type type,
+                  Type resultType, list<Trait> traits = []> :
+    LLVM_Op<mnemonic, !listconcat([Pure], [DeclareOpInterfaceMethods<IntegerOverflowFlagsInterface>], traits)>,
+    LLVM_Builder<"$res = builder.Create" # instName # "($arg, $_resultType, /*Name=*/\"\", op.hasNoUnsignedWrap(), op.hasNoSignedWrap());"> {
+  let arguments = (ins type:$arg, EnumProperty<"IntegerOverflowFlags", "", "IntegerOverflowFlags::none">:$overflowFlags);
+  let results = (outs resultType:$res);
+  let builders = [LLVM_OneResultOpBuilder];
+  let assemblyFormat = "$arg attr-dict `` custom<OverflowFlags>($overflowFlags) `:` type($arg) `to` type($res)";
```

Comment:
```suggestion
  let assemblyFormat = "$arg custom<OverflowFlags>($overflowFlags) `` attr-dict `:` type($arg) `to` type($res)";
```
nit: Can you move the attribute dictionary after the overflow flags. I think this is how it is done for the other operations that take overflow flags. 

I am somewhat surprised you don't need to elide overflow flags from the attribute dictionary when printing using `custom<LLVMOpAttrs>(attr-dict)`. Maybe tablegen does elide the flag since it is passed to a custom print function.

Reply:
Since the integer overflow flags are properties they should not be printed as part of attr-dict and `custom<LLVMOpAttrs>(attr-dict)` is not needed. In fact, it seems like the print/parseLLVMOpAttrs only makes sense for the fast math flags these days.

Reply:
I'm not sure what you mean by "`custom<LLVMOpAttrs>(attr-dict)` is not needed". Does anything need to be changed besides the nit above?

Reply:
Changing the nit should be sufficient (especially since the tests pass). The purpose of `custom<LLVMOpAttrs>(attr-dict)` is to filter out specific attributes from the attribute dictionary. This is not needed anymore now that the flag is a property.

Reply:
That worked, thank you